components/s3/id.ftl

```diff
@@ -0,0 +1,146 @@
+[#ftl]
+[@addResourceGroupInformation
+    type=S3_COMPONENT_TYPE
+    attributes=
+        [
+            {
+                "Names" : "Lifecycle",
+                "Children" : [
+                    {
+                        "Names" : "BlobRetentionDays",
+                        "Type" : NUMBER_TYPE,
+                        "Default" : ""
+                    },
+                    {
+                        "Names" : "BlobAutoSnapshots",
+                        "Type" : BOOLEAN_TYPE,
+                        "Default" : false
+                    }
+                ]
+            },
+            {
+                "Names" : "Website",
+                "Children" : [
+                    {
+                        "Names" : "CustomDomain",
+                        "Type" : STRING_TYPE,
+                        "Default" : ""
+                    },
+                    {
+                        "Names" : "HttpsOnly",
+                        "Description" : "Support Only Https traffic?",
+                        "Type" : BOOLEAN_TYPE,
+                        "Default" : true
+                    }
+                ]
+            },
+            {
+                "Names" : "Encryption",
+                "Children" : [
+                    {
+                        "Names" : "Enabled",
+                        "Type" : BOOLEAN_TYPE,
+                        "Default" : false
+                    },
+                    {
+                        "Names" : "KeySource",
+                        "Type" : STRING_TYPE,
+                        "Values" : [ "Microsoft.Storage", "Microsoft.Keyvault" ],
+                        "Default" : "Microsoft.Keyvault"
+                    },
+                    {
+                        "Names" : "Services",
+                        "Type" : ARRAY_OF_STRING_TYPE,
+                        "Description" : "The Services to enable encryption for. Valid values are 'blob' and/or 'file'.",
+                        "Default" : [ "" ]
+                    }
+                ]
+            },
+            {
+                "Names" : "Secrets",
+                "Children" : [
+                    {
+                        "Names" : "KeyName",
+                        "Type" : STRING_TYPE,
+                        "Default" : ""
+                    },
+                    {
+                        "Names" : "KeyVersion",
+                        "Type" : STRING_TYPE,
+                        "Default" : ""
+                    },
+                    {
+                        "Names" : "KeyUri",
+                        "Type" : STRING_TYPE,
+                        "Default" : ""
+                    }
+                ]
+            },
+            {
+                "Names" : "Access",
+                "Children" : [
+                    {
+                        "Names" : "SubnetIds",
```

Comment:
IP Address group lookup as an example is here in the load balancer 
https://github.com/codeontap/gen3/blob/fece5bf524683e50f007aaf9982383d11b4c55d1/providers/aws/components/lb/setup.ftl#L195 

You can skip the public route table lookup and just pass a string array of the IP Address group names

Reply:
Removing these attributes to use existing functions [this is my latest change](https://github.com/codeontap/gen3-azure/pull/23/commits/52166bc62cad1ce77701d1d08a47c9a745506d64). 

The subnetId needs to be the Azure Reference Id, which will look something like "/subscriptions/{subscriptionId}/resourceGroups/{groupName}/providers/Microsoft.Network/virtualNetworks/{vnetName}/subnets/{subnetName}". This will then need to be an output on the network resource which will follow soon.